buildAndReleaseTask/index.ts

```diff
@@ -1,100 +1,205 @@
-        tl.cp(linuxBin, appknoxPath, "-f");
-        return fs.chmodSync(appknoxPath, "755");
+    const url = getAppknoxDownloadURL(os);
+
+    const tmpDir = 'binaries'
+    mkdirp.sync(tmpDir);
+    const tmpFile = path.join(__dirname, tmpDir, supportedOS[os].name);
+
+    tl.debug(`Downloading appknox binary from ${url}`);
+    await downloadFile(url, proxy, tmpFile);
+
+    if (!fs.existsSync(tmpFile)) {
+        throw Error("Could not download appknox binary");
     }
-    throw Error("unsupported os " + os);
+
+    tl.debug("Download finished");
+
+    supportedOS[os].copyToBin(tmpFile, "755");
+    tl.debug(`Appknox installation completed: ${supportedOS[os].path}`);
+
+    return supportedOS[os].path;
 }
 
+
 async function upload(filepath: string, riskThreshold: string) {
-    tl.debug("Filepath: " + filepath);
-    tl.debug("Riskthreshold: " + riskThreshold);
-    await copyAppknox();
-    const appknoxPath = getAppknoxPath();
-    tl.debug("AppknoxBinaryPath: " + appknoxPath);
-    try {
-        const appknoxUploader = tl.tool(appknoxPath);
-        const xargs = tl.tool('xargs')
-        appknoxUploader.arg("upload")
-                        .arg(filepath)
-                        .arg("--access-token")
-                        .arg(token);
-        xargs.arg(appknoxPath)
-                        .arg("cicheck")
-                        .arg("--risk-threshold")
-                        .arg(riskThreshold)
-                        .arg("--access-token")
-                        .arg(token);
-        appknoxUploader.pipeExecOutputToTool(xargs);
-        const ret = await appknoxUploader.exec();
-        return ret;
+    tl.debug(`Filepath: ${filepath}`);
+    tl.debug(`Riskthreshold: ${riskThreshold}`);
+
+    const _execOptions = <trm.IExecOptions>{
+        silent: false,
+        failOnStdErr: true,
     }
-    catch(err){
+
+    try {
+        const proxy = getProxyURL();
+        const hasValidProxy = isValidURL(proxy);
+        const appknoxPath = await installAppknox(os, proxy);
+
+        const uploadCmd: trm.ToolRunner = tl.tool(appknoxPath);
+        uploadCmd.arg("upload")
+            .arg(filepath)
+            .arg("--access-token")
+            .arg(token)
+            .argIf(hasValidProxy, "--proxy")
+            .argIf(hasValidProxy, proxy);
+
+        let result: trm.IExecSyncResult = uploadCmd.execSync(_execOptions);
+        if (result.code == 1) {
+            throw result.error;
+        }
+        const fileID: string = result.stdout.trim();
+
+        const checkCmd: trm.ToolRunner = tl.tool(appknoxPath);
+        checkCmd.arg("cicheck")
+            .arg(fileID)
+            .arg("--risk-threshold")
+            .arg(riskThreshold)
+            .arg("--access-token")
+            .arg(token)
+            .argIf(hasValidProxy, "--proxy")
+            .argIf(hasValidProxy, proxy);
+        return await checkCmd.exec(_execOptions);
+
+    } catch(err) {
         tl.setResult(tl.TaskResult.Failed, err.message);
```

Comment:
also log it

Reply:
`setResult` internally logs error message: https://github.com/microsoft/azure-pipelines-task-lib/blob/master/node/task.ts#L77-L95